Execution failed for task ':tools:go-cli:distBinary'

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. If there is a duplicate, please close your issue and add a comment
to the existing issue.

If you suspect your issue is a bug, please edit your issue description to
include the information shown below.

For more information about reporting issues, see
https://github.com/openwhisk/openwhisk/blob/master/CONTRIBUTING.md#raising-issues


Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a feature request
-->

## Environment details:

* vagrant 1.8.1   
*   OS Name:                   Microsoft Windows 7 Professional
* OS Version:                6.1.7601 Service Pack 1 Build 7601
* terminal : bash shell


## Steps to reproduce the issue:

1. git clone --depth=1 https://github.com/openwhisk/openwhisk.git
2. cd openwhisk/tools/vagrant
3. ./hello

## Provide the actual results and outputs:

```
output comes here

==> default: Successfully built 63bb4d1e5956
==> default: :tools:go-cli:tagImage
==> default: Executing 'docker tag -f whisk/go-cli whisk/go-cli:latest'
==> default: :tools:go-cli:pushImage
==> default:
==> default: SKIPPED
==> default: :tools:go-cli:removeBinary
==> default: :tools:go-cli:distBinary
==> default: Executing 'docker rm -f go-cli'
==> default: Executing 'mkdir /home/vagrant/openwhisk/tools/go-cli/../../bin/go-                                                                      cli/'
==> default: Executing 'docker run --name go-cli whisk/go-cli:latest'
==> default: Executing 'docker cp go-cli:/src/github.com/go-whisk-cli/build/. /h                                                                      ome/vagrant/openwhisk/tools/go-cli/../../bin/go-cli'
==> default: ln: failed to create symbolic link ‘/home/vagrant/openwhisk/bin/go-                                                                      cli/wsk’: Protocol error
==> default: :tools:go-cli:distBinary FAILED
==> default: FAILURE:
==> default: Build failed with an exception.
==> default: * Where:
==> default: Build file '/home/vagrant/openwhisk/tools/go-cli/build.gradle' line                                                                      : 31
==> default: * What went wrong:
==> default: Execution failed for task ':tools:go-cli:distBinary'.
==> default: >
==> default: Process 'command 'ln'' finished with non-zero exit value 1
==> default: * Try:
==> default: Run with
==> default: --stacktrace
==> default:  option to get the stack trace. Run with
==> default: --info
==> default:  or
==> default: --debug
==> default:  option to get more log output.
==> default: BUILD FAILED
==> default: Total time: 7 mins 58.614 secs
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.



## Additional information you deem important:
* issue happens only occasionally or under certain circumstances   
* changes you did or observed in the environment

--- a/tools/go-cli/build.gradle
+++ b/tools/go-cli/build.gradle
@@ -2,6 +2,43 @@ ext.dockerImageName = "whisk/go-cli"
 ext.dockerContainerName = "go-cli"
 apply from: '../../gradle/docker.gradle'
 
+
+task removeBinary(type: Delete) {
+    delete "${projectDir}/../../bin/go-cli"
+}
+
+task distBinary(dependsOn: [removeBinary, distDocker]) << {
+    run(dockerBinary + ["rm", "-f", dockerContainerName], true)
+    run(dockerBinary + ["run", "--name", dockerContainerName, dockerRegistry + dockerImageName + ":" + dockerImageTag])
+
+    // Copy all Go binaries from Docker into openwhisk/bin/go-cli folder
+    run(dockerBinary + ["cp", dockerContainerName +
+            ":/src/github.com/go-whisk-cli/build/.", "${projectDir}/../../bin/go-cli"])
+
+    run(dockerBinary + ["rm", "-f", dockerContainerName])
+}
+
+task dumpOSInfo {
+    println "os.name = "+getOsName()
+    println "os.arch = "+getOsArch()
+    println "go.name = "+mapOsNameToGoName(getOsName())
+    println "go.arch = "+mapOsArchToGoArch(getOsArch())
+}
+
+task copyCLIShortcut(type: Copy, dependsOn: [distBinary, dumpOSInfo]) {
+    String go_osname     = mapOsNameToGoName(getOsName())
+    String go_osarch     = mapOsArchToGoArch(getOsArch())
+    String from_path_wsk = "${projectDir}/../../bin/go-cli/${go_osname}/${go_osarch}/wsk"
+    String to_path_dir   = "${projectDir}/../../bin/go-cli"
+
+    from from_path_wsk
+    into to_path_dir
+ }
+
+
+
+distDocker.finalizedBy copyCLIShortcut
+
 // Returns the Go CLI docker build args
 def getDockerBuildArgs() {
     java.text.SimpleDateFormat sdf = new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ")
@@ -10,34 +47,6 @@ def getDockerBuildArgs() {
     return "BUILDTIME="+dateStr
 }
 
-task removeBinary(type: Delete) {
-    delete "${projectDir}/../../bin/go-cli"
-}
-
-task distBinary(dependsOn: [removeBinary, distDocker]) << {
-    run(dockerBinary + ["rm", "-f", dockerContainerName], true)
-    run(["mkdir", "${projectDir}/../../bin/go-cli/"])
-    run(dockerBinary + ["run", "--name", dockerContainerName, dockerRegistry + dockerImageName + ":" + dockerImageTag])
-
-    // Copy all Go binaries from Docker into openwhisk/bin/go-cli folder
-    run(dockerBinary + ["cp", dockerContainerName +
-            ":/src/github.com/go-whisk-cli/build/.", "${projectDir}/../../bin/go-cli"])
-
-    // Create a shorthand sym link to current platform appropriate Go CLI binary
-    String go_osname = mapOsNameToGoName(getOsName())
-    String go_osarch = mapOsArchToGoArch(getOsArch())
-    String link_to_path = new File("${projectDir}/../../bin/go-cli/wsk").getCanonicalPath()
-    String link_from_path = new File("${projectDir}/../../bin/go-cli/${go_osname}/${go_osarch}/wsk").getCanonicalPath()
-    exec {
-        executable "ln"
-        args "-s", "${link_from_path}", "${link_to_path}"
-    }
-
-    run(dockerBinary + ["rm", "-f", dockerContainerName])
-}
-
-distDocker.finalizedBy([removeBinary, distBinary])
-
 def run(cmd, ignoreError = false) {
     println("Executing '${cmd.join(" ")}'")
     def proc = cmd.execute()
@@ -45,13 +54,6 @@ def run(cmd, ignoreError = false) {
     if(!ignoreError && proc.exitValue() != 0) {
         println("Command '${cmd.join(" ")}' failed with exitCode ${proc.exitValue()}")
     }
-}
-
-task dump_os() << {
-    println "os.name = "+getOsName()
-    println "os.arch = "+getOsArch()
-    println "go.name = "+mapOsNameToGoName(getOsName())
-    println "go.arch = "+mapOsArchToGoArch(getOsArch())
 }
 
 def getOsName() {

